keras_cv/models/resnet_v2.py

```diff
@@ -325,90 +331,142 @@ def ResNetV2(
 
-    x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="post_bn")(x)
-    x = layers.Activation("relu", name="post_relu")(x)
+        x = layers.MaxPooling2D(3, strides=2, padding="same", name="pool1_pool")(x)
+
+        num_stacks = len(stackwise_filters)
+        if stackwise_dilations is None:
+            stackwise_dilations = [1] * num_stacks
+
+        stack_level_outputs = {}
+        for stack_index in range(num_stacks):
+            x = Stack(
+                x,
+                filters=stackwise_filters[stack_index],
+                blocks=stackwise_blocks[stack_index],
+                stride=stackwise_strides[stack_index],
+                dilations=stackwise_dilations[stack_index],
+                block_fn=block_fn,
+                first_shortcut=block_fn == Block or stack_index > 0,
+                stack_index=stack_index,
+            )
+            stack_level_outputs[stack_index + 2] = x
 
-    if include_top:
-        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
-        x = layers.Dense(classes, activation=classifier_activation, name="predictions")(
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=BN_EPSILON, name="post_bn")(
             x
         )
-    else:
-        if pooling == "avg":
-            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
-        elif pooling == "max":
-            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+        x = layers.Activation("relu", name="post_relu")(x)
 
-    # Create model.
-    model = tf.keras.Model(inputs, x, name=name, **kwargs)
-
-    if weights is not None:
-        model.load_weights(weights)
-    # Set this private attribute for recreate backbone model with outputs at each of the
-    # resolution level.
-    model._backbone_level_outputs = stack_level_outputs
-
-    # Bind the `to_backbone_model` method to the application model.
-    model.as_backbone = types.MethodType(utils.as_backbone, model)
-
-    return model
+        if include_top:
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+            x = layers.Dense(
+                classes, activation=classifier_activation, name="predictions"
+            )(x)
+        else:
+            if pooling == "avg":
+                x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+            elif pooling == "max":
+                x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+        # Create model.
+        super().__init__(inputs=inputs, outputs=x, name=name, **kwargs)
+
+        # All references to `self` below this line
+        if weights is not None:
+            self.load_weights(weights)
+        # Set this private attribute for recreate backbone model with outputs at each of the
+        # resolution level.
+        self._backbone_level_outputs = stack_level_outputs
+
+        # Bind the `to_backbone_model` method to the application model.
+        self.as_backbone = types.MethodType(utils.as_backbone, self)
```

Comment:
Why not have a specific `def as_backbone` method on the class?

Reply:
If it's OK with you I'd like do handle this in a followup PR where I create a `Backbone` base class to hold the method as the long term solution.